diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3e47224..c4e3fca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,9 @@ set(NULL_ENTROPY_WARNING "${WARNING_BORDER}"
                          "${NULL_ENTROPY_WARN_L3}"
                          "${WARNING_BORDER}")
 
-find_package(Perl)
+#find_package(Perl)
+set(PERL_EXECUTABLE "perl")
+
 if(PERL_FOUND)
 
     # If NULL Entropy is configured, display an appropriate warning
